torchaudio/functional/filtering.py

```diff
@@ -957,7 +962,11 @@ def lfilter(
     assert a_coeffs.ndim <= 2
 
     if a_coeffs.ndim > 1:
-        waveform = torch.stack([waveform] * a_coeffs.shape[0], -2)
+        if batching:
+            assert waveform.ndim > 1
+            assert waveform.shape[-2] == a_coeffs.shape[0]
+        else:
+            waveform = torch.stack([waveform] * a_coeffs.shape[0], -2)
```

Comment:
This shape manipulation looks very complicated, and I wonder if it is because we have `pack/unpack batch` in the following section. If so, is there a way to contain `pack/unpack batch` bellow into `batching` mechanism?

Reply:
I think it's not related to `pack/unpack batch`. 
Waveform and coefficients should have equal number of channels before pack batch, and these conditions make sure this behavior.

Reply:
We can move the nested conditions outside though

```python
    if a_coeffs.ndim == 1:
        a_coeffs = a_coeffs.unsqueeze(0)
        b_coeffs = b_coeffs.unsqueeze(0)

    if batching:
        assert waveform.ndim > 1
        assert waveform.shape[-2] == a_coeffs.shape[0]
    else:
        waveform = torch.stack([waveform] * a_coeffs.shape[0], -2)

    # pack batch
```